packages/backend/server/src/core/doc-service/service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import {
+  Config,
+  CryptoHelper,
+  InternalServerError,
+  UserFriendlyError,
+} from '../../base';
+import { PgWorkspaceDocStorageAdapter } from '../doc';
+import { type DocRecord } from '../doc/storage';
+
+@Injectable()
+export class DocService {
+  private readonly logger = new Logger(DocService.name);
+
+  constructor(
+    private readonly config: Config,
+    private readonly workspace: PgWorkspaceDocStorageAdapter,
+    private readonly crypto: CryptoHelper
+  ) {}
+
+  async getDoc(workspaceId: string, docId: string): Promise<DocRecord | null> {
+    if (!this.config.docService.useRpc) {
```

Comment:
Could be implemented as dynamic provider with simple strategy.

```
class DocRader {
  get() {
    // read db
  }
}

class RpcDocReader extends DocReader {
   override get() {
     // rpc call
   }
}

const DocReaderProvider: FactoryProvider = {
  provide: DocReader,
  useFactory(config, ref) => {
    config.rpc ? ref.create(RpcDocReader) : ref.create(DocReader)
  },
  inject: [Config, ModuleRef]
}

@Module({
  providers: [DocReaderProvider],
  exports: [DocReader],
})
class DocReaderModule {}

class DocService {
  constructor(reader: DocRenderer) {}

  get() {
    return this.reader.get()
  }
}

```

Reply:
https://docs.nestjs.com/fundamentals/custom-providers#custom-providers-1